tests/composer/test_blueprints.py

```diff
@@ -173,20 +173,20 @@ def test_changes(self):
     # NOTE: Order of these 3 is important, delete needs to come after save, before push and these 3
     @unittest.skipUnless(os.path.exists("/run/weldr/api.socket"), "Test requires a running API server")
     def test_save_0(self):
-        """blueprints save example-development"""
+        """blueprints save"""
         blueprints_save("/run/weldr/api.socket", 0, ["example-http-server"], show_json=False)
         self.assertTrue(os.path.exists("example-http-server.toml"))
 
     @unittest.skipUnless(os.path.exists("/run/weldr/api.socket"), "Test requires a running API server")
     def test_save_1(self):
-        """blueprints delete example-development"""
+        """blueprints delete"""
         rc = blueprints_delete("/run/weldr/api.socket", 0, ["example-development"], show_json=False)
```

Comment:
Don't you have to delete `example-http-server` which is now referenced above and below ? 

@bcl I don't quite understand the behavior of these 3 test methods and why their order matters. Looks fragile to me. Can you explain what is the intended behavior here ?

Reply:
If I delete `example-http-server` then the last test called `blueprints-diff` fails because it won't find this blueprint. But I guess Brian would know better what's exactly happening here I was also confused when trying to figure this out.

Reply:
Yes, these are fragile because they are running against an actual API server so the changes persist and may effect later test results, that's why I used different blueprints for some of them, so that the changes wouldn't cause problems with example-http-server which is used for most of them.

If we need to add another test blueprint to replace glusterfs or whatever that's fine with me but the changes here should be as minimal as possible.

Reply:
@jrusz from what I can tell by Brian's comment you shouldn't have to make changes here. As of now I'm not sure we know this piece of test code well enough to be confident that these changes result in the test methods exercising the same functionality that was tested before the change. 

If it did work in Travis then you shouldn't have to change anything here or am I missing something ?

Reply:
Well that's because these tests never ran in Travis, they were just skipped. I first encountered this when trying to run `make ci` here on master branch. I fixed those tests but it never made it to rhel8-branch because there were missing packages in epel8 and we didn't want to use pip. Now that we know we have to use pip regardless I decided to go back to this and make it work.

Reply:
ok, I see what's happening now. In Travis there was no running server that's why these test methods were skipped. 

Regardless of that, some of the blueprints should still be available. I see `example-development.toml` and `example-glusterfs.toml` under `./tests/pylorax/blueprints`. Is this the directory used by the server API while these tests are running ? IOW are we starting the API server from the git checkout or relying on the system wide installation to be running via systemd ? This is what I still can't understand.

Reply:
> IOW are we starting the API server from the git checkout or relying on the system wide installation to be running via systemd ? This is what I still can't understand.

After some investigation I believe that this depends on whether or not is the `$CLI` set. 
The following is from `tests/test_cli.sh`
```
 82 if [ -z "$CLI" ]; then
 83     export top_srcdir=`pwd`
 84     . ./tests/testenv.sh
 85 
 86     export BLUEPRINTS_DIR=`mktemp -d '/tmp/composer-blueprints.XXXXX'`
 87     cp ./tests/pylorax/blueprints/*.toml $BLUEPRINTS_DIR
 88 
 89     export SHARE_DIR=`mktemp -d '/tmp/composer-share.XXXXX'`
 90     cp -R ./share/* $SHARE_DIR
 91     chmod a+rx -R $SHARE_DIR
 92 
 93     setup_tests $SHARE_DIR $BLUEPRINTS_DIR
 94     # start the lorax-composer daemon
 95     composer_start
 96 else
 97     export PACKAGE="composer-cli"
 98     export BLUEPRINTS_DIR="/var/lib/lorax/composer/blueprints"
 99     composer_stop
100     setup_tests /usr/share/lorax /var/lib/lorax/composer/blueprints
101     composer_start
102 fi

```
So since I don't set `$CLI` in the test I would say this runs with the git checkout.

Reply:
So in this case we're starting the lorax server from the git checkout and instruct it to serve the blueprints which are available in the source code. There definitely is an `example-development.toml` file in there so all of these changes shouldn't be necessary. 

From what I can tell the sequence of steps is: upload a blueprint and make sure it exists; then delete it and make sure it doesn't exist and finally upload it again, maybe as a means to restore the environment like it was before.

Reply:
Right that seems to be what's happening here, but if I leave all this with `example-development` another test fails and I get this:
```
======================================================================                                                                                                                                                                                                          
FAIL: blueprints diff                                                                                                                                                                                                                                                           
----------------------------------------------------------------------                                                                                                                                                                                                          
Traceback (most recent call last):                                                                                                                                                                                                                                              
  File "/lorax/tests/composer/test_blueprints.py", line 274, in test_z_diff                                                                                                                                                                                                     
    self.assertTrue("Changed Version" in output)                                                                                        
AssertionError: False is not true                                                                                                                                                                                                                                               
-------------------- >> begin captured logging << --------------------                                                                                                                                                                                                          
urllib3.connectionpool: DEBUG: http://localhost:None "GET /api/v0/blueprints/changes/example-http-server?limit=0 HTTP/1.1" 200 163                                                                                                                                              
urllib3.connectionpool: DEBUG: http://localhost:None "GET /api/v0/blueprints/changes/example-http-server?limit=1 HTTP/1.1" 200 401                                                                                                                                              
urllib3.connectionpool: DEBUG: http://localhost:None "GET /api/v0/blueprints/diff/example-http-server/14c14d18d3713c909e68e03026ba5a0869528a5b/HEAD HTTP/1.1" 200 17                                                                                                            
--------------------- >> end captured logging << ---------------------                                                                                                   
```
But I can make it pass by making changes in that particular test and the one above it. So this seems to be a question of what we want to change, I can change either and this will pass... 

@bcl could you please decide which do you prefer? I can see these options:

1. Leave it as it is now, meaning have the `test_save_0, test_save_1, test_save_2` use `example-http-server`
1. Have those tests use `example-development` and then change `test_workspace, test_z_diff` to use `example-development` as well